include/openthread/platform/misc.h

```diff
@@ -207,6 +207,14 @@ otError otPlatSetMcuPowerState(otInstance *aInstance, otPlatMcuPowerState aState
  */
 otPlatMcuPowerState otPlatGetMcuPowerState(otInstance *aInstance);
 
+/**
+ * Logs a crash dump using OpenThread logging APIs
+ *
+ * @retval OT_ERROR_NONE            Crash dump was logged successfully
+ * @retval OT_ERROR_NOT_CAPABLE     Platform is not capable of logging a crash dump
+ */
+otError otPlatLogCrashDump(void);
```

Comment:
Does adding this as a `otPlat` API make sense?
- `otPlat` APIs are intended for OT core modules use.
-  We don't use this from any OT API.
-  It is used in `ncp_base.cpp` and from `main` (CLI)